LocalOrbit/localorbit
# [LO-1045] Stripe signed webhooks configuration

Update stripe_event gem to support signed webhooks requests from Stripe.

- Refactor and clean up webhook specs.
- Update vcr config
- fixed bug where existing payments that later failed were not sync'ing the failed state

TODO:
- [ ] Weston: rebase on master, add some `dead_code!` alerts into payment mailer and `transfer.paid` webhook handler
- [ ] Rob: update the stripe plans in `stripe.rake`
- [ ] Figure out why/where/when transfers are being sent with regards to Connect
- [ ] figure out how to invoke an invoice.payment_succeeded event
- [ ] figure out how to invoke an invoice.payment_failed event

Deployment:
- [x] configure staging with it's own Stripe account
- [x] configure `STRIPE_ACCOUNT_SIGNING_SECRET` env var on `staging`
- [x] configure `STRIPE_CONNECT_SIGNING_SECRET` env var on `staging`
- [x] configure `STRIPE_ACCOUNT_SIGNING_SECRET` env var on `production`
- [x] configure `STRIPE_CONNECT_SIGNING_SECRET` env var on `production`

To test on dev/staging:
- [ ] use `rake stripe:create_plans` task to create plans and watch for webhooks to succeed
- create a new plan (product) via rails console (to ensure API version) and verify it
  appears in the plans database table
- create a successful invoice
- test the OAuth flow for Stripe Connect as we updated the related oauth gems
- test the buying flows as we updated stripe gem
